src/main/java/broker/LogSegment.java

```diff
@@ -26,7 +32,7 @@ public LogSegment(int partitionNumber, int startOffset) {
             } else {
                 Logger.warn(String.format("File %s already exists", this.logFile.getPath()));
             }
-        } catch (IOException e) {
+        } catch (IOException | NullPointerException e) {
```

Comment:
Out of curiosity, are we expecting a NPE here?

Reply:
im tweaking i actually dont remember why i included NPE here 🤔

Reply:
>I should probably keep it as an IOException?

Yea that's what I think, but at the same time, i feel like you may not even need to catch any exception and instead let them bubble up

Reply:
> instead let them bubble up

so just have the constructor `throw Exception` or something like that instead of the try-catch block?